src/Session.js

```diff
@@ -38,14 +38,24 @@ const Session = class Session {
         });
     }
 
-    markAccessed(modelName) {
-        this.getDataForModel(modelName).accessed = true;
+    markAccessed(modelName, modelIds) {
+        const data = this.getDataForModel(modelName);
+        if (!data.accessed) {
+            data.accessed = [];
+        }
+        data.accessed.push(...modelIds);
     }
 
     get accessedModels() {
         return this.sessionBoundModels
             .filter(model => !!this.getDataForModel(model.modelName).accessed)
-            .map(model => model.modelName);
+            .reduce(
```

Comment:
This reduce strikes me as a little clunky; I mostly wanted to retain a similar structure to the function as it used to be, but we could probably just examine the `getDataForModel` results instead of combing through `this.sessionBoundModels`.